src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/TableRowProjection.java

```diff
@@ -0,0 +1,85 @@
+package io.stargate.sgv2.jsonapi.service.operation.tables;
+
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.metadata.schema.ColumnMetadata;
+import com.datastax.oss.driver.api.querybuilder.select.Select;
+import com.datastax.oss.driver.api.querybuilder.select.SelectFrom;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.operation.DocumentSource;
+import io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs.JSONCodec;
+import io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs.JSONCodecRegistry;
+import io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs.MissingJSONCodecException;
+import io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs.ToJSONCodecException;
+import io.stargate.sgv2.jsonapi.service.projection.TableProjectionDefinition;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Projection used for Table Rows (as opposed to Collection Documents), built from command API
+ * projection definitions (expressed in JSON).
+ */
+public record TableRowProjection(
+    ObjectMapper objectMapper, TableSchemaObject table, List<ColumnMetadata> columns)
+    implements OperationProjection {
+  /**
+   * Factory method for construction projection instance, given a projection definition and table
+   * schema.
+   */
+  public static TableRowProjection fromDefinition(
+      ObjectMapper objectMapper,
+      TableProjectionDefinition projectionDefinition,
+      TableSchemaObject table) {
+    Map<String, ColumnMetadata> columnsByName = new HashMap<>();
+    table
+        .tableMetadata
+        .getColumns()
+        .forEach((id, column) -> columnsByName.put(id.asInternal(), column));
+
+    List<ColumnMetadata> columns = projectionDefinition.extractSelectedColumns(columnsByName);
+
+    if (columns.isEmpty()) {
```

Comment:
A table can't be with empty columns. Think a redundant check.

Reply:
Left a TODO, just in case it breaks anything

Reply:
We are not checking whether table has no columns but whether set of columns specified by projection definition matching actual columns is empty.
But if validity of projection columns was performed, this could indeed be redundant (problem caught elsewhere).